gax-java/gax/src/main/java/com/google/api/gax/rpc/ClientContext.java

```diff
@@ -346,6 +353,11 @@ private static Map<String, String> getHeadersFromSettings(StubSettings settings)
     effectiveHeaders.putAll(userHeaders);
     effectiveHeaders.putAll(conflictResolution);
 
+    if (credentialTypeForMetrics != CredentialTypeForMetrics.DO_NOT_SEND) {
+      effectiveHeaders.computeIfPresent(
+          ApiClientHeaderProvider.getDefaultApiClientHeaderKey(),
```

Comment:
Is it decided to append `cred-type/` to `x-goog-api-client` header? It seems all other header tokens in `x-goog-api-client` are static values from the client, not from user input. I wonder why we didn't choose to use a separate header key for this purpose?

Reply:
yes, this design is already implemented for other languages. See details in design [go/googleapis-auth-metric-design](http://goto.google.com/googleapis-auth-metric-design) section 2.3.2. I don't think I saw discussion on using a different header specifically though.

Reply:
I see your point. 
Let me know if you see better suited places for additional docs.